### PR TITLE
Pin SQLAlchemy below 2

### LINKED ISSUE #250: Sqlalchemy v2.0.0 bug causes `cells.SingleCells` to not work: ObjectNotExecutableError with read_sql
I noticed that when I pip installed Pycytominer to use in a project yesterday that I was getting this error when trying to run `cells.SingleCells` from the [cells.py file](https://github.com/cytomining/pycytominer/blob/master/pycytominer/cyto_utils/cells.py). 

```python
ObjectNotExecutableError: Not an executable object: 'select * from Per_Image'
```

This error is caused by a bug where between Pandas and the newest release of Sqlalchemy from a few days ago. It is described further in [this issue](https://github.com/pandas-dev/pandas/issues/51061) from the Pandas GitHub.

I noticed that in the [requirements.txt](https://github.com/cytomining/pycytominer/blob/master/requirements.txt) that it states:

```txt
sqlalchemy>=1.3.6
```
This means it is installing the latest version of Sqlalchemy when pip installing. This means a new user if they wanted to use the `cells.SingleCells` function or any function in Pycytominer that uses Sqlalchemy `read_sql` function using a string, then it will not work.

The only work around I have that works is a conda environment that installs Sqlalchemy v1.4.42 (which is the last version that works) and pip installs Pycytominer. The conda install override the pip install. 

Once this bug is fixed, then this won't be a worry, but I wanted to make this issue for others to be aware in case they get this same error.

### PR DESCRIPTION
**Pull request by bethac07**
_modified from [EmbeddedArtistry](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/)_

# Description

Resolves #250

## What is the nature of your change?

- [x] Bug fix (fixes an issue).

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: requirements.txt
Comment (gwaybio) [inline requirements.txt:5]:
```suggestion
sqlalchemy<=1.4.42
```

Do you think it makes sense to be less restrictive? If you disagree, I will merge with a pinned version

**Review by gwaybio — APPROVED** [review]

File: requirements.txt
Comment (bethac07) [reply requirements.txt:5]: I'd recommend going with `sqlalchemy>=1.3.6,<2`, or `sqlalchemy~=1.4.42` if you don't want an exact pin - I assume the previous version minimum was for a reason. But your call.

Comment (gwaybio) [reply requirements.txt:5]: @d33bs - what do you recommend?

Comment (d33bs) [reply requirements.txt:5]: Thanks @gwaybio - I would recommend what @bethac07 mentioned with a version range here, respecting that any broken aspects of the SQLAlchemy API in v2 may incur separate updates for a time. Specifically, I'd vote for `sqlalchemy>=1.3.6,<2` to help make sure we account for any reasons why 1.3.6 was preferred at one point or another. Dependencies could be updated and tested otherwise (for ex. updating >= 1.4.42) outside of the focus of this PR and #250 .

Comment (bethac07) [reply requirements.txt]: Cool, have updated to @d33bs 's preference, since already approved will pull when tests are done!

**Comment by gwaybio** [comment]
Thanks @bethac07 and @d33bs !

**Pull request merged**
